sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ProjectingInternalRow.scala

```diff
@@ -116,3 +117,9 @@ case class ProjectingInternalRow(schema: StructType, colOrdinals: Seq[Int]) exte
     row.get(colOrdinals(ordinal), dataType)
   }
 }
+
+object ProjectingInternalRow {
+  def apply(schema: StructType, colOrdinals: Seq[Int]): ProjectingInternalRow = {
+    new ProjectingInternalRow(schema, colOrdinals.toIndexedSeq)
```

Comment:
Do you have benchmark number? cc @yaooqinn

Reply:
This seems like a classic ["accidentally quadratic"](https://accidentallyquadratic.tumblr.com/) O(n^2) issue so you could achieve pretty arbitrary speedups with a contrived benchmark.

From the flamegraph attached to https://issues.apache.org/jira/browse/SPARK-49408 it seems like this would make a pretty substantial difference for that test query: it looks like ~94% of the time was being spent in list traversal.